src/query/api/v1/handler/prometheus/native/middleware.go

```diff
@@ -23,84 +23,52 @@ package native
 
 import (
 	"net/http"
-	"strings"
 	"time"
 
-	"github.com/gorilla/mux"
-	"github.com/jonboulle/clockwork"
 	"go.uber.org/zap"
 
-	"github.com/m3db/m3/src/query/util/logging"
-	"github.com/m3db/m3/src/x/headers"
-	xhttp "github.com/m3db/m3/src/x/http"
-	"github.com/m3db/m3/src/x/instrument"
+	"github.com/m3db/m3/src/query/api/v1/middleware"
 )
 
-// QueryResponseOptions are the options to construct the QueryResponse middleware func.
-type QueryResponseOptions struct {
-	Threshold      time.Duration
-	InstrumentOpts instrument.Options
-	Clock          clockwork.Clock
+// WithQueryParams adds the query request parameters to the middleware options.
+var WithQueryParams middleware.OverrideOptions = func(opts middleware.Options) middleware.Options {
```

Comment:
the query response logging and non-query response logging were merged into a single middleware func that does all response logging. this has the added benefit of improving the non-query response logging with things we only added to the query logging, like status code.

now the query endpoints need to register this option to add the query params as log fields. while I could have hacked and just looked for `query`, this seemed cleaner. the `middleware` pkg should know nothing about the `native` pkg.

Reply:
additionally this override style is used for the metrics middleware as well for similar reasons. in order to extract the query params from the request.